lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionScope.kt

```diff
@@ -838,11 +719,11 @@ public interface ExpressionScope {
    * zoom 24, it is 256. Applied to for example line width, this has the visual effect that the line
    * stays the same width in meters on the map (rather than on the viewport).
    */
-  public fun <Output> interpolate(
-    type: Expression<TInterpolationType>,
-    input: Expression<Number>,
-    vararg stops: Pair<Number, Expression<Output>>,
-  ): Expression<Output> = interpolateImpl("interpolate", type, input, *stops)
+  public fun <Output : Expression.Interpolateable> interpolate(
```

Comment:
example of a properly typed expression now _without_ adding a bunch of overloads